java/configuration/src/main/java/sleeper/configuration/table/index/DynamoDBTableIndex.java

```diff
@@ -148,75 +165,132 @@ public void delete(TableStatus table) {
                                 .withTableName(idIndexDynamoTableName)
                                 .withKey(DynamoDBTableIdFormat.getIdKey(table))
                                 .withConditionExpression("attribute_exists(#tableid)")
-                                .withExpressionAttributeNames(Map.of("#tableid", TABLE_ID_FIELD))));
+                                .withExpressionAttributeNames(Map.of("#tableid", TABLE_ID_FIELD))),
+                        new TransactWriteItem().withDelete(new Delete()
+                                .withTableName(onlineIndexDynamoTableName)
+                                .withKey(DynamoDBTableIdFormat.getOnlineKey(table))
+                                .withConditionExpression("attribute_exists(#tablename)")
+                                .withExpressionAttributeNames(Map.of("#tablename", TABLE_NAME_FIELD))));
         try {
             TransactWriteItemsResult result = dynamoDB.transactWriteItems(request);
             List<ConsumedCapacity> consumedCapacity = result.getConsumedCapacity();
             double totalCapacity = consumedCapacity.stream().mapToDouble(ConsumedCapacity::getCapacityUnits).sum();
             LOGGER.debug("Deleted table {}, capacity consumed = {}", table, totalCapacity);
         } catch (TransactionCanceledException e) {
-            CancellationReason nameNotFoundReason = e.getCancellationReasons().get(0);
-            CancellationReason idNotFoundReason = e.getCancellationReasons().get(1);
-            if (isCheckFailed(nameNotFoundReason)) {
+            int nameDeleteIndex = 0;
+            int idDeleteIndex = 1;
+            int onlineDeleteIndex = 2;
+            if (isCheckFailed(e, nameDeleteIndex)) {
                 throw TableNotFoundException.withTableName(table.getTableName());
-            } else if (isCheckFailed(idNotFoundReason)) {
+            }
+            if (isCheckFailed(e, idDeleteIndex)) {
                 throw TableNotFoundException.withTableId(table.getTableUniqueId());
-            } else {
-                throw e;
             }
+            if (isCheckFailed(e, onlineDeleteIndex)) {
+                throw TableNotFoundException.withTableName(table.getTableName());
+            }
```

Comment:
These can keep the original exception as the cause of the TableNotFoundException.